client/src/pages/SeriesDetailsPage.js

```diff
@@ -0,0 +1,128 @@
+// import PropTypes from 'prop-types'
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router'
+import { Link } from 'react-router-dom'
+import styled from 'styled-components/macro'
+import Poster from '../components/Poster'
+import IconLeftArrow from '../components/ui/IconLeftArrow'
+
+export default function SeriesDetails() {
+  const { id } = useParams()
+  const [seriesDetails, setSeriesDetails] = useState([])
+  const [cast, setCast] = useState([])
+
+  useEffect(() => {
+    fetch(`/api/series/${id}`)
+      .then(res => res.json())
+      .then(data => setSeriesDetails(data))
+      .catch(error => {
+        console.error('Error:', error)
+      })
+  }, [id])
+
+  useEffect(() => {
+    fetch(`/api/series/${id}/credits`)
+      .then(res => res.json())
+      .then(data => setCast(data.cast))
+      .catch(error => {
+        console.error('Error:', error)
+      })
+  }, [id])
+
+  const { name, poster_path: posterPath, overview } = seriesDetails
+
+  return (
+    <Wrapper>
+      <BackButton to="/">
+        <IconLeftArrow />
+      </BackButton>
+      <Header>
+        <Poster
+          path={
+            posterPath !== undefined
+              ? `https://image.tmdb.org/t/p/w300/${posterPath}`
+              : '../poster.png'
+          }
```

Comment:
I think at this point the loading is a bit slow or problematic? When I´m clicking on one Film first the poster.png is shown and some seconds later the right image of the film. Maybe this issue is caused here? Maybe it needs to long to look for the right poster path?

Reply:
Hey, good point, thank you! I noticed that as well. Locally the problem did not happen because it's probably fast enough. I'm already working on a solution.